java11/src/test/java/feign/httpclient/test/JavaHttp2ClientTest.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright 2012-2018 The Feign Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package feign.httpclient.test;
+
+import feign.*;
+import feign.httpclient.JavaHttp2Client;
+import org.assertj.core.api.Assertions;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Tests client-specific behavior, such as ensuring Content-Length is sent when specified.
+ */
+public class JavaHttp2ClientTest {
+
+  public interface TestInterface {
+    @RequestLine("POST /?foo=bar&foo=baz&qux=")
+    @Headers({"Foo: Bar", "Foo: Baz", "Qux: ", "Content-Type: text/plain"})
+    Response post(String var1);
+
+    @RequestLine("POST /path/{to}/resource")
+    @Headers({"Accept: text/plain"})
+    Response post(@Param("to") String var1, String var2);
+
+    @RequestLine("GET /")
+    @Headers({"Accept: text/plain"})
+    String get();
+
+    @RequestLine("PATCH /patch")
+    @Headers({"Accept: text/plain"})
+    String patch(String var1);
+
+    @RequestLine("POST")
+    String noPostBody();
+
+    @RequestLine("PUT")
+    String noPutBody();
+
+    @RequestLine("POST /?foo=bar&foo=baz&qux=")
+    @Headers({"Foo: Bar", "Foo: Baz", "Qux: ", "Content-Type: {contentType}"})
+    Response postWithContentType(String var1, @Param("contentType") String var2);
+  }
+
+  @Test
+  public void testPatch() throws Exception {
+    TestInterface api = newBuilder().target(TestInterface.class, "https://nghttp2.org/httpbin/");
+    Assertions.assertThat(api.patch(""))
+        .contains("https://nghttp2.org/httpbin/patch");
```

Comment:
Won't this make a real call?  Is there a way we can mock this out to prevent builds from failing due to network or host issues?

Reply:
I wanted to be sure it was hitting an http2 server....
No network in general means no build for maven =)

Do you know any http2 mocks?